vpr/src/route/route_timing.h

```diff
@@ -10,9 +10,196 @@
+    //
+    // This is used when you want to allow previous routing of the same net to
+    // serve as valid start locations for the current connection.
+    //
+    // Returns either the last element of the path, or nullptr if no path is
+    // found
+    t_heap* timing_driven_route_connection_from_route_tree(
+        t_rt_node* rt_root,
+        int sink_node,
+        const t_conn_cost_params cost_params,
+        t_bb bounding_box,
+        RouterStats& router_stats);
+
+    // Finds a path from the route tree rooted at rt_root to sink_node for a
+    // high fanout net.
+    //
+    // Unlike timing_driven_route_connection_from_route_tree(), only part of
+    // the route tree which is spatially close to the sink is added to the heap.
+    t_heap* timing_driven_route_connection_from_route_tree_high_fanout(
+        t_rt_node* rt_root,
+        int sink_node,
+        const t_conn_cost_params cost_params,
+        t_bb bounding_box,
+        const SpatialRouteTreeLookup& spatial_rt_lookup,
+        RouterStats& router_stats);
+
+    // Finds a path from the route tree rooted at rt_root to all sinks
+    // available.
+    //
+    // Each element of the returned vector is a reachable sink.
+    //
+    // If cost_params.astar_fac is set to 0, this effectively becomes
+    // Dijkstra's algorithm with a modified exit condition (runs until heap is
+    // empty).  When using cost_params.astar_fac = 0, for efficiency the
+    // RouterLookahead used should be the NoOpLookahead.
+    std::vector<t_heap> timing_driven_find_all_shortest_paths_from_route_tree(
+        t_rt_node* rt_root,
+        const t_conn_cost_params cost_params,
+        t_bb bounding_box,
+        RouterStats& router_stats);
+
+  private:
+    // Mark that inode has been modified, and needs to be reset in
+    // reset_path_costs.
+    void add_to_mod_list(int inode) {
+        if (std::isinf(rr_node_route_inf_[inode].path_cost)) {
+            modified_rr_node_inf_.push_back(inode);
+        }
+    }
+
+    // Update the route path to the node pointed to by cheapest.
+    inline void update_cheapest(t_heap* cheapest) {
+        update_cheapest(cheapest, &rr_node_route_inf_[cheapest->index]);
+    }
+
+    inline void update_cheapest(t_heap* cheapest, t_rr_node_route_inf* route_inf) {
+        //Record final link to target
+        add_to_mod_list(cheapest->index);
+
+        route_inf->prev_node = cheapest->u.prev.node;
+        route_inf->prev_edge = cheapest->u.prev.edge;
+        route_inf->path_cost = cheapest->cost;
+        route_inf->backward_path_cost = cheapest->backward_path_cost;
+    }
+
+    // Common logic from timing_driven_route_connection_from_route_tree and
+    // timing_driven_route_connection_from_route_tree_high_fanout for running
+    // connection router.
+    t_heap* timing_driven_route_connection_from_route_tree(
+        t_rt_node* rt_root,
+        int sink_node,
+        const t_conn_cost_params cost_params,
+        t_bb bounding_box);
```

Comment:
I think this should have a different name (its currently overloaded with the public method). Perhaps something like `timing_driven_route_connection_common_setup()`?

Reply:
Done